arch/xtensa/Kconfig

```diff
@@ -73,6 +73,7 @@ config XTENSA_SMALL_VECTOR_TABLE_ENTRY
 config XTENSA_CACHED_REGION
 	int "Cached RPO mapping"
 	range 0 7
+	default 5
```

Comment:
This is going to leak into all Xtensa devices, when the value is specific to the intel_adsp devices (actually just cAVS, ACE has different regions).  Now, as it happens only the intel_adsp boot code actually interprets this, but it's not impossible that we'll unify the boot path in the future and trip over this.  Might be safer to wrap both inside a "CONFIG_XTENSA_RPO_CACHE_TRICK" or something.

Reply:
it was already leaking :)

The intel adsp settings were being used by all Xtensa targets although nobody is actually using those cache functions referencing the values, so nothing really broke.
I started by moving this to each target, but that did not make sense to add it to every single targrt, so I reverted the defaults above.

But I guess I need one more config guard so we do not build this code if not being used...